include/nbl/builtin/hlsl/tonemapper/operators.hlsl

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_TONE_MAPPER_OPERATORS_INCLUDED_
+#define _NBL_BUILTIN_HLSL_TONE_MAPPER_OPERATORS_INCLUDED_
+
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+#include "nbl/builtin/hlsl/type_traits.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace tonemapper
+{
+
+template<typename T = float32_t>
+struct Reinhard
+{
+	using float_t = enable_if_t<is_floating_point<T>::value, T>;
+	using float_t3 = typename conditional<is_same_v<float_t, float32_t>, float32_t3, float16_t3>::type;
+	using this_t = Reinhard<float_t>;
+	static this_t create(float_t EV, float_t key = 0.18f, float_t WhitePointRelToEV = 16.f)
+	{
+		this_t retval;
+		retval.keyAndManualLinearExposure = key * exp2(EV);
+		retval.rcpWhite2 = 1.f / (WhitePointRelToEV * WhitePointRelToEV);
+		return retval;
+	}
+
+	float_t3 operator()(float_t3 rawCIEXYZcolor) {
+		float_t exposureFactors = keyAndManualLinearExposure;
+		float_t exposedLuma = rawCIEXYZcolor.y * exposureFactors;
+		float_t colorMultiplier = (exposureFactors * (1.0 + exposedLuma * rcpWhite2) / (1.0 + exposedLuma));
```

Comment:
the 1.0 needs to be dealt with in a funny way by making sure its the correct type constant 
```
const float_t unit = 1.0;
```
because in C++ a literal 1.0 is a double and in HLSL its a float32_t

Reply:
Done